### PR TITLE
Support hex strings as byte input args

### PR DESCRIPTION
**Pull request by ggarri**
## Changes
Support String() refect types to be used for `encodeBytes()` and `encodeFixedBytes()` to prevent panic exceptions

## Resolved issue
https://github.com/umbracle/go-web3/issues/149

### CODE REVIEW

File: abi/encode.go
Comment (ferranbt) [inline abi/encode.go:209]: Could you do this instead, so we can reuse the `packBytesSlice` call?
```suggestion
		v = value
```

**Review by ferranbt — CHANGES_REQUESTED** [review]

File: abi/encode.go
Comment (ggarri) [reply abi/encode.go:209]: It cannot since `v` is expected as `reflect.Value` and `value` is a `[]byte`. I would do the mapping but IMO that would be an unnecessary overhead in the code. WDYT?

Comment (ferranbt) [reply abi/encode.go]: You are right on that, I like your approach better.

Comment (ggarri) [reply abi/encode.go]: Hi @ferranbt if you refresh you should already see that change applied in my PR. 

PD: We are waiting for them to be merged in master to fix an issue on our software affecting ours users.

Comment (ferranbt) [reply abi/encode.go]: Ok, as soon as CI is done I merge it

**Review by ferranbt — APPROVED** [review]

**Pull request merged**
